cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguageFrontend.kt

```diff
@@ -62,21 +64,32 @@ class PythonLanguageFrontend(language: Language<PythonLanguageFrontend>, ctx: Tr
      * new [PythonLanguageFrontend] instance per file.
      */
     private lateinit var fileContent: String
-    private lateinit var uri: URI
+    private var filePath: Path? = null
 
     @Throws(TranslationException::class)
     override fun parse(file: File): TranslationUnitDeclaration {
-        fileContent = file.readText(Charsets.UTF_8)
-        uri = file.toURI()
+        return parse(file.readText(Charsets.UTF_8), file.toPath())
+    }
+
+    override fun parse(content: String, path: Path?): TranslationUnitDeclaration {
+        this.fileContent = content
+        this.filePath = path
 
         jep.getInterp().use {
-            it.set("content", fileContent)
-            it.set("filename", file.absolutePath)
+            it.set("content", content)
+            it.set(
+                "filename",
+                if (path != null) {
+                    path.absolute().toString()
+                } else {
+                    "<unknown>"
```

Comment:
To we want this constant for cpg consistency? Python recommends `<string>` see https://docs.python.org/3/library/functions.html#compile

Reply:
We are using ast.Parse and the default parameter here was `<unknown>`. See https://docs.python.org/3/library/ast.html#ast.parse

Reply:
> Funny, that the ast.parse doc says:
> 
> > Equivalent to compile(source, filename, mode, PyCF_ONLY_AST).
> 
> But I'm ok with the "unknown". This should probably a sane constant for all frontends if there is no file available?

I was intentionally using the python specific value here, but I am fine with either way. Not sure if the SARIF standard defines something in this case, since we are using the location information from SARIF.